modules/store/spec/action_with_modifiers.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { ActionCreator, TypedAction } from '../src/models';
+import { createActionWithModifiers } from '../src/action_with_modifiers_creator';
+import { props } from '../src/action_creator';
+
+describe('createActionWithModifiers', () => {
+  it('should inject properties onto action', () => {
+    // given
+    const notificationMessageKey = 'notification-message';
+    const hasNotification = <P>(config: { message: string }) => <
+      T extends string
+    >(
+      action: ActionCreator<T, (p: P) => TypedAction<T> & P>
+    ) => {
+      const creator = (p: P) => ({
+        ...action(p),
+        [notificationMessageKey]: config.message,
+      });
+      creator.type = action.type;
+      return creator;
+    };
+
+    const requestIdKey = 'request-id';
+    const isEndOfRequest = <P>(config: { requestState: { id: number } }) => <
+      T extends string
+    >(
+      action: ActionCreator<T, (p: P) => TypedAction<T> & P>
+    ) => {
+      const creator = (p: P) => ({
+        ...action(p),
+        [requestIdKey]: config.requestState.id,
+      });
+      creator.type = action.type;
+      return creator;
+    };
+
+    const createRequestState = () => ({ id: Math.random() });
+
+    const aFormRequestState = createRequestState();
+
+    const aFormSaveSuccess = createActionWithModifiers(
+      'A Form Save Success',
+      props<{ form: { name: string } }>(),
+      (payload) =>
+        hasNotification({
+          message: `${payload.form.name} has been saved as the new name`,
+        }),
+      () => isEndOfRequest({ requestState: aFormRequestState })
+    );
```

Comment:
I was thinking the API could look more like this, but when I tried writing the helper functions I wasn't able to get the type inference to work correctly:

```typescript
const aFormSaveSuccess = createActionWithModifiers(
  'A Form Save Success',
  props<{ form: { name: string } }>(),
  is(({ form })=> aNotification({ message: `${form.name} has been saved as the new name` })),
  is(() => anEndOfRequest({ requestState: aFormRequestState }))
);
```